js/task-controller.js

```diff
@@ -0,0 +1,104 @@
+export default class TaskController {
+
+    _tasks = [];
+
+    currentTasks_container = document.querySelector("#currentTasks");
+    completedTasks_container = document.querySelector("#completedTasks");
+
+    setTasks(newTasks) {
+        this._tasks = newTasks;
+    }
+
+    getTasks() {
+        return this._tasks;
+    }
+
+    addTask(task) {
+        const newTasks = this.getTasks();
+        newTasks.push(task);
+        this.setTasks(newTasks);
+    }
+
+    updateTask(props) {
+        const currentTasks = this.getTasks();
+
+        const { index, title, text, priority } = props;
+        currentTasks[index].title = title;
+        currentTasks[index].text = text;
+        currentTasks[index].priority = priority;
+
+        this.setTasks(currentTasks);
+    }
+
+    completeTask(id) {
+        const currentTasks = this.getTasks();
+
+        const taskIndex = currentTasks.findIndex(task => task.id == id);
+        currentTasks[taskIndex].isCompleted = !currentTasks[taskIndex].isCompleted;
+        this.setTasks(currentTasks);
+    }
+
+    deleteTask(id) {
+
+        const currentTasks = this.getTasks();
+        const taskIndex = currentTasks.findIndex(task => task.id == id);
+
+        const newTasks = [
+            ...currentTasks.slice(0, taskIndex),
+            ...currentTasks.slice(taskIndex + 1)
+        ];
```

Comment:
it's simpler with `filter` method

Reply:
I like one line coding :sunglasses:
```suggestion
        const newTasks = currentTasks.filter(task => task.id != id);
```